src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9714,7 +9714,7 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
         /// have the same signature, ignoring parameter names and custom modifiers. The particular
         /// method returned is not important since the caller is interested in the signature only.
         /// </summary>
-        private MethodSymbol? GetUniqueSignatureFromMethodGroup(BoundMethodGroup node)
+        private MethodSymbol? GetUniqueSignatureFromMethodGroup_CSharp10(BoundMethodGroup node)
```

Comment:
>GetUniqueSignatureFromMethodGroup_CSharp10 [](http://example.com/codeflow?start=30&length=42)

Is the difference between the two implementations essentially just the `if (method is not null)` check to break early from the `SearchExtensionMethods` loop? If so, consider sharing the implementation between the two language versions so it's clear what the difference is, and to (hopefully) simplify maintaining the implementations.

Reply:
Let's discuss after LDM next week. If we go with the broader change for handling of generics (which I have in a follow-up PR), the two methods will diverge a lot more